### PR TITLE
chore: Save package statistics to DB

### PR DESCRIPTION
**Pull request by miroslavstastny**
Part of #773.

Each CI build stores results of `yarn build:stats` to an external DB.
Following data are stored:
- build number,
- GIT commit SHA,
- GIT branch,
- GIT tag (to filter release builds),
- GitHub PR URL,
- build date
- current bundle size statistics.

No magic is currently done on DB side, data is stored as-is. Based on the required queries, we can restructure the data and migrate all existing ones.

No user-facing change, changelog left intentionally blank.

### CODE REVIEW

**Comment by miroslavstastny** [comment]
This is repo for the DB API endpoint: https://github.com/miroslavstastny/stardust-stats-azure (to be moved to stardust repo later)

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "release:patch": "yarn prerelease && ta-script npm/release patch && yarn postrelease",
     "prestart": "yarn satisfied --fix yarn",
     "start": "gulp --series dll docs",
+    "stats:save": "gulp stats:save",
```
Comment (layershifter) [inline package.json:37]: Can we rename `build:stats` to `stats:build`? (line 5)

```diff
-"build:stats": "gulp stats",
+"stats:build": "gulp stats:build",
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -89,7 +97,7 @@ jobs:
       - run:
           name: Publish npm package
           command: |
-            if [ -n "${npm_TOKEN+x}" ] && [ "${CIRCLE_BRANCH}" == "master" ]; then
+            if [ -n "${npm_TOKEN}" ] && [ "${CIRCLE_BRANCH}" == "master" ]; then
```
Comment (layershifter) [inline .circleci/config.yml:95]: This change looks strange, is it correct?

File: build/gulp/tasks/stats.ts [hunk 3]
```diff
@@ -121,3 +123,27 @@ task(
...
+task('stats:save', async () => {
+  const commandLineArgs = _.pick(argv, ['sha', 'branch', 'tag', 'pr', 'build'])
+  const bundleStats = require(paths.docsSrc('bundleStats.json'))[UNRELEASED_VERSION_STRING]
+
+  const options = {
+    method: 'POST',
+    uri: process.env.STATS_URI,
+    body: {
+      sha: process.env.CIRCLE_SHA1,
+      branch: process.env.CIRCLE_BRANCH,
```
Comment (layershifter) [inline build/gulp/tasks/stats.ts]: There values are required as I understand, can we add an assert that they're defined?

File: .circleci/config.yml (hunk 2 above)
Comment (kuzhelov) [reply .circleci/config.yml:95]: yes, as `-n` `test` function already correctly handles the values `+x` filter has targeted on

File: build/gulp/tasks/stats.ts [hunk 4]
```diff
@@ -121,3 +123,27 @@ task(
...
+      pr: process.env.CIRCLE_PULL_REQUEST, // optional
+      build: process.env.CIRCLE_BUILD_NUM,
+      ...commandLineArgs, // allow command line overwrites
+      bundleSize: bundleStats,
+      ts: new Date(),
+    },
+    json: true,
+  }
+
+  const ret = await rp(options)
```
Comment (kuzhelov) [inline build/gulp/tasks/stats.ts]: lets omit shortcuts here, as it makes the code a bit unreadable. We could end up with something like (with import aliasing and variable being renamed)

```js
const response = await requestHttp(options)
```

**Review by layershifter — APPROVED** [review]
👍

**Review by kuzhelov — APPROVED** [review]

File: .circleci/config.yml [hunk 5]
```diff
@@ -74,7 +74,15 @@ jobs:
...
           name: Bundle Statistics
-          command: yarn build:stats
+          command: yarn stats:build
+      - run:
+          name: Save Statistics to DB
+          command: |
+            if [ -n "${STATS_URI}" ]; then
+              yarn stats:save
+            else
+              echo "STATS_URI not set, skipping"
```
Comment (levithomason) [inline .circleci/config.yml:84]: Seems this should throw, no?  Do we want silent passes when stats aren't being saved for some reason?

**Pull request merged**